CMakeLists.txt

```diff
@@ -1,551 +1,565 @@
-    target_compile_options(devilution PUBLIC -fno-omit-frame-pointer)
-  endif()
-  # Ignore serious warnings due to "quality" of decompiled code
-  # Currently, disable ignore all warnings (-w), to be removed later
-  target_compile_options(devilution PRIVATE -fpermissive -w)
-
-  # Warnings for devilutionX
-  target_compile_options(${BIN_TARGET} PRIVATE -Wall -Wextra -Wno-write-strings -Wno-unused-parameter -Wno-missing-field-initializers -Wno-format-security)
-
-  # For ARM and other default unsigned char platforms
-  target_compile_options(devilution PRIVATE -fsigned-char)
-  target_compile_options(${BIN_TARGET} PRIVATE -fsigned-char)
-endif()
-
-if(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-  target_compile_options(devilution PRIVATE "/W0")
-endif()
-
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  # Style issues
-  target_compile_options(${BIN_TARGET} PRIVATE -Wno-parentheses -Wno-logical-op-parentheses -Wno-bitwise-op-parentheses)
-  # Silence warnings about __int64 alignment hack not always being applicable
-  target_compile_options(${BIN_TARGET} PRIVATE -Wno-ignored-attributes)
-  # Silence appfat.cpp warnings
-  set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-narrowing")
-endif()
-
-if(APPLE)
-  set_source_files_properties("./Packaging/macOS/AppIcon.icns" PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
-  set_source_files_properties("./Packaging/resources/CharisSILB.ttf" PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
-  set(MACOSX_BUNDLE_GUI_IDENTIFIER com.diasurgical.devilutionx)
-  set(MACOSX_BUNDLE_COPYRIGHT Unlicense)
-  set(MACOSX_BUNDLE_BUNDLE_NAME devilutionx)
-  set(MACOSX_BUNDLE_INFO_STRING ${PROJECT_VERSION})
-  set(MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION})
-  set(MACOSX_BUNDLE_SHORT_VERSION_STRING ${PROJECT_VERSION})
-  set(MACOSX_BUNDLE_LONG_VERSION_STRING "Version ${PROJECT_VERSION}")
-  set_target_properties(${BIN_TARGET} PROPERTIES MACOSX_BUNDLE_ICON_FILE "AppIcon")
-  set_target_properties(${BIN_TARGET} PROPERTIES MACOSX_BUNDLE_INFO_PLIST "${CMAKE_CURRENT_SOURCE_DIR}/Packaging/macOS/Info.plist")
-
-  install (TARGETS ${BIN_TARGET} DESTINATION ./)
-
-  if(DIST)
-      install(CODE "
-        include(BundleUtilities)
-        fixup_bundle(${CMAKE_BINARY_DIR}/${MACOSX_BUNDLE_BUNDLE_NAME}.app \"\" \"\")
-        "
-        COMPONENT Runtime)
-  endif()
-
-  set(MACOSX_BUNDLE_LONG_VERSION_STRING "Version ${PROJECT_VERSION}")
-  set(CPACK_PACKAGE_FILE_NAME "devilutionx")
-  set(CPACK_DMG_DISABLE_APPLICATIONS_SYMLINK "ON")
-  set(CPACK_GENERATOR "DragNDrop")
-
-  include(CPack)
-endif()
-
-if(SWITCH)
-  set(APP_TITLE "DevilutionX")
-  set(APP_AUTHOR "Devilution Team")
-  set(APP_ICON "${PROJECT_SOURCE_DIR}/Packaging/switch/icon.jpg")
-  set(APP_VERSION ${PROJECT_VERSION})
-  include(nx-utils)
-  build_switch_binaries(${BIN_TARGET})
-endif()
-
-if(RETROFW OR RG350)
-  set_target_properties(${BIN_TARGET} PROPERTIES OUTPUT_NAME "devilutionx.dge")
-endif()
+cmake_minimum_required(VERSION 3.8) # CMP0083 NEW
```

Comment:
Please fix the newline on this, as now it looks like the whole file has changed

Reply:
I will reuse the original file then. All source is "suffering" from "dos cr": Apart from that, using some editors the whole source has big spacing between line codes. Maybe I will use another editor.

Reply:
the line endings is converted by git on Windows if you don't tell it to ignore it. It's a pain in the ass. I've configured my computers to always keep line endings as is when checking out to avoid this.

Reply:
Old school CygnusEd and others see all that but not the hidden characters: Even some of my changes were marked in "red" by github. Not good.

Reply:
It using CR since we expect that to have been the case for the original, it's-not-a-bug-it's-a-feature :)